presto-product-tests/src/main/java/com/facebook/presto/tests/ImpersonationTests.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.tests;
+
+import com.facebook.presto.tests.ImmutableTpchTablesRequirements.ImmutableNationTable;
+import com.google.inject.Inject;
+import com.google.inject.name.Named;
+import com.teradata.tempto.ProductTest;
+import com.teradata.tempto.Requires;
+import com.teradata.tempto.hadoop.hdfs.HdfsClient;
+import org.testng.annotations.Test;
+
+import static com.facebook.presto.tests.TestGroups.PROFILE_SPECIFIC_TESTS;
+import static com.facebook.presto.tests.TestGroups.SINGLENODE;
+import static com.facebook.presto.tests.TestGroups.SINGLENODE_HDFS_IMPERSONATION;
+import static com.facebook.presto.tests.TestGroups.SINGLENODE_KERBEROS_HDFS_IMPERSONATION;
+import static com.facebook.presto.tests.TestGroups.SINGLENODE_KERBEROS_HDFS_NO_IMPERSONATION;
+import static com.teradata.tempto.query.QueryExecutor.query;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+import static org.testng.Assert.assertEquals;
+
+@Requires(ImmutableNationTable.class)
+public class ImpersonationTests
+        extends ProductTest
+{
+    @Inject
+    private HdfsClient hdfsClient;
+    private static final String tableName = "checkTableOwner";
```

Comment:
- Extract `/user/hive/warehouse` as a `tempto-configuration.yaml` property, called say `databases.hive.warehouse_directory_path`. That property should be set to `/user/hive/warehouse` by default, but it must be configurable, cause it is configurable in hive.
- Use different table name for every single test, and leave those tables alive if the test fails.
- Instead of hard-coding table path better create some private method, called `getTableLocation(String tableName)`, and use the `databases.hive.warehouse_directory_path` within.